go/pkg/sysdb/grpc/collection_service.go

```diff
@@ -570,6 +572,131 @@ func (s *Server) FlushCollectionCompaction(ctx context.Context, req *coordinator
+		log.Error("FlushCollectionCompactionAndAttachedFunction failed. flush_compactions is empty")
+		return nil, grpcutils.BuildInternalGrpcError("at least one flush_compaction is required")
+	}
+	if len(flushReqs) > 2 {
+		log.Error("FlushCollectionCompactionAndAttachedFunction failed. too many flush_compactions", zap.Int("count", len(flushReqs)))
+		return nil, grpcutils.BuildInternalGrpcError("expected 1 or 2 flush_compactions")
+	}
+
+	// Parse attached function update info
+	attachedFunctionUpdate := req.GetAttachedFunctionUpdate()
+	if attachedFunctionUpdate == nil {
+		log.Error("FlushCollectionCompactionAndAttachedFunction failed. attached_function_update is nil")
+		return nil, grpcutils.BuildInternalGrpcError("attached_function_update is required")
+	}
+
+	attachedFunctionID, err := uuid.Parse(attachedFunctionUpdate.Id)
+	if err != nil {
+		log.Error("FlushCollectionCompactionAndAttachedFunction failed. error parsing attached_function_id", zap.Error(err), zap.String("attached_function_id", attachedFunctionUpdate.Id))
+		return nil, grpcutils.BuildInternalGrpcError("invalid attached_function_id: " + err.Error())
+	}
+
+	// Validate completion_offset fits in int64 before storing in database
+	if attachedFunctionUpdate.CompletionOffset > uint64(math.MaxInt64) {
+		log.Error("FlushCollectionCompactionAndAttachedFunction: completion_offset too large",
+			zap.Uint64("completion_offset", attachedFunctionUpdate.CompletionOffset))
+		return nil, grpcutils.BuildInternalGrpcError("completion_offset too large")
+	}
+	completionOffsetSigned := int64(attachedFunctionUpdate.CompletionOffset)
+
+	// Parse all flush requests into a slice
+	collectionCompactions := make([]*model.FlushCollectionCompaction, 0, len(flushReqs))
+
+	for _, flushReq := range flushReqs {
+		collectionID, err := types.ToUniqueID(&flushReq.CollectionId)
+		err = grpcutils.BuildErrorForUUID(collectionID, "collection", err)
+		if err != nil {
+			log.Error("FlushCollectionCompactionAndAttachedFunction failed. error parsing collection id", zap.Error(err), zap.String("collection_id", flushReq.CollectionId))
+			return nil, grpcutils.BuildInternalGrpcError(err.Error())
+		}
+
+		segmentCompactionInfo := make([]*model.FlushSegmentCompaction, 0, len(flushReq.SegmentCompactionInfo))
+		for _, flushSegmentCompaction := range flushReq.SegmentCompactionInfo {
+			segmentID, err := types.ToUniqueID(&flushSegmentCompaction.SegmentId)
+			err = grpcutils.BuildErrorForUUID(segmentID, "segment", err)
+			if err != nil {
+				log.Error("FlushCollectionCompactionAndAttachedFunction failed. error parsing segment id", zap.Error(err), zap.String("collection_id", flushReq.CollectionId))
+				return nil, grpcutils.BuildInternalGrpcError(err.Error())
+			}
+			filePaths := make(map[string][]string)
+			for key, filePath := range flushSegmentCompaction.FilePaths {
+				filePaths[key] = filePath.Paths
+			}
+			segmentCompactionInfo = append(segmentCompactionInfo, &model.FlushSegmentCompaction{
```

Comment:
Do we properly handle empty compactions and the like here?

Reply:
yes, it's the same as normal compactions. That case is tested by compact.rs::test_compaction_with_empty_logs_from_inserts_and_deletes